my-webxr-app/src/utils/PCA-classic.tsx

```diff
@@ -0,0 +1,58 @@
+import { Matrix, SVD } from 'ml-matrix';
+import assert from "./assert.tsx";
+import {standardizeDataset} from "./standardizeDataset.tsx"
+
+/**
+ * Inspired by: https://jonathan-hui.medium.com/machine-learning-singular-value-decomposition-svd-principal-component-analysis-pca-1d45e885e491
+ * 
+ * Steps involved in this method of PCA
+ *    1: Standardize the dataset.
+ *    2: Calulate the k-th right singular vectors by performming singular value decomposition on dataset.
+ *    3: Take product of dataset and k-th right singular vectors to form PCA matrix.
+ */
+
+/**
+ * Perform SVD on a dataset and get the first k right singular vectors.
+ * 
+ * @param {Matrix} datasetMatrix - The dataset to perform SVD on.
```

Comment:
This will perform PCA on whatever 2D matrix of the dataset we pass in. So if we just want pca on some column of the dataset, then we just truncate it and pass it here. I won't pass in the whole CSV because there could be non-numeric data.